test/integration/apiserver/cel/validatingadmissionpolicy_test.go

```diff
@@ -0,0 +1,937 @@
+						Namespace: "default",
+					},
+					PolicyName: "validate-namespace-suffix",
+				},
+			},
+			configMap: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "validate-namespace-suffix-param",
+					Namespace: "default",
+				},
+				Data: map[string]string{
+					"namespaceSuffix": "k8s",
+				},
+			},
+			namespace: &v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-foo",
+				},
+			},
+			err:           "ValidatingAdmissionPolicy 'validate-namespace-suffix' with binding 'validate-namespace-suffix-binding' denied request: failed expression: object.metadata.name.endsWith(params.data.namespaceSuffix)",
+			failureReason: metav1.StatusReasonInvalid,
+		},
+	}
+
+	for _, testcase := range testcases {
+		t.Run(testcase.name, func(t *testing.T) {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.CELValidatingAdmission, true)()
+			server, err := apiservertesting.StartTestServer(t, nil, []string{
+				"--enable-admission-plugins", "ValidatingAdmissionPolicy",
+			}, framework.SharedEtcd())
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer server.TearDownFn()
+
+			config := server.ClientConfig
+
+			client, err := clientset.NewForConfig(config)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if _, err := client.CoreV1().ConfigMaps("default").Create(context.TODO(), testcase.configMap, metav1.CreateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
+			if _, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicies().Create(context.TODO(), testcase.policy, metav1.CreateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
+			if _, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicyBindings().Create(context.TODO(), testcase.policyBinding, metav1.CreateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
+			// TODO: add retry logic instead
+			time.Sleep(time.Second)
+
+			_, err = client.CoreV1().Namespaces().Create(context.TODO(), testcase.namespace, metav1.CreateOptions{})
+			if err == nil && testcase.err == "" {
+				return
+			}
+
+			if err == nil && testcase.err != "" {
+				t.Logf("actual error: %v", err)
+				t.Logf("expected error: %v", testcase.err)
+				t.Fatal("got nil error but expected an error")
+			}
+
+			if err != nil && testcase.err == "" {
+				t.Logf("actual error: %v", err)
+				t.Logf("expected error: %v", testcase.err)
+				t.Fatal("got error but expected none")
+			}
+
+			if err.Error() != testcase.err {
+				t.Logf("actual validation error: %v", err)
+				t.Logf("expected validation error: %v", testcase.err)
+				t.Error("unexpected validation error")
+			}
+
+			reason := err.(apierrors.APIStatus).Status().Reason
+			if reason != testcase.failureReason {
+				t.Logf("actual error reason: %v", reason)
+				t.Logf("expected failure reason: %v", testcase.failureReason)
+				t.Error("unexpected error reason")
+			}
+		})
+	}
+}
+
+func TestNoResource(t *testing.T) {
```

Comment:
Can we use a better name for this tests?

Reply:
`Test_ValidatingAdmissionPolicy_ParamObjectDeleted`?

Reply:
Seems like the test needs to be updated to remove the ConfigMap though, do you want to just remove this test for now and add it as a follow-up?

Reply:
Yeah, I think we can remove the test from this PR for now - I'll mark this as resolved when I do that.

At the moment I'm still in favor of the test getting implemented with [this TODO](https://github.com/cici37/kubernetes/pull/9#discussion_r1014375606) though that's a separate concern I think.